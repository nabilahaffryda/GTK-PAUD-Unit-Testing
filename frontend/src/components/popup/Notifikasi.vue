<template>
  <v-dialog
    v-model="dialog"
    :max-width="options.width ? options.width : 650"
    persistent
    scrollable
    transition="dialog-bottom-transition"
  >
    <v-card>
      <v-card-title class="pa-0" v-if="!(options && options.custom)">
        <v-toolbar dark :color="options.toolbarColor || 'secondary'" flat>
          <v-toolbar-title v-html="options.title || title"></v-toolbar-title>
          <v-spacer />
          <v-btn icon dark @click="onClose" v-if="!(options && options.persist)">
            <v-icon>mdi-close</v-icon>
          </v-btn>
        </v-toolbar>
      </v-card-title>
      <v-card-text :class="[options.custom ? 'border-top-studi' : '']">
        <slot name="content">
          <v-row class="my-5" dense no-gutters>
            <v-col cols="2" v-if="!options.noIcon">
              <v-avatar :color="options.avatarColor || '#FFE6E6'" size="64">
                <v-icon :color="options.iconColor || 'error'">
                  {{ options.icon || `mdi-alert` }}
                </v-icon>
              </v-avatar>
            </v-col>
            <v-col :cols="options.noIcon ? 12 : 10">
              <div v-html="message"></div>
            </v-col>
          </v-row>
        </slot>
      </v-card-text>
      <template v-if="(options && options.useBtn) || !((options && options.persist) || false)">
        <v-divider></v-divider>
        <v-card-actions>
          <slot name="action">
            <v-spacer></v-spacer>
            <v-btn :color="notif.color || '#26a69a'" dark @click="onAction" v-if="notif && notif.action">
              {{ notif.button }}
            </v-btn>
            <v-btn class="ma-2" :color="options.btnColor || '#26a69a'" dark @click="onClose" v-else>
              {{ (options && options.btnLabel) || 'OK' }}
            </v-btn>
          </slot>
        </v-card-actions>
      </template>
    </v-card>
  </v-dialog>
</template>

<script>
export default {
  name: 'PopupNotifikasi',
  data() {
    return {
      dialog: false,
      title: 'Pengumuman',
      message: '',
      options: {},
    };
  },
  props: {
    notif: {
      type: Object,
      default: () => {},
    },
  },
  methods: {
    open(message, title, options) {
      this.title = title || this.title;
      this.message = message || this.message;
      this.options = options || this.options;
      this.dialog = true;
    },
    onClose() {
      this.dialog = false;
      if (this.options && this.options.action) this.options && this.options.action();
    },
    onAction() {
      this.$nextTick(() => {
        this.dialog = false;
        this.$emit('action', this.notif.action);
      });
    },
  },
};
</script>

<style scoped>
.border-top-studi {
  border-top: 7px solid #00bfa5;
}
</style>
