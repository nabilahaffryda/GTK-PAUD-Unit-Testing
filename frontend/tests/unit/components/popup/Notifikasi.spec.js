import { mount, createLocalVue } from "@vue/test-utils";
import Notifikasi from "@/components/popup/Notifikasi.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';

Vue.use(Vuetify)
const localVue = createLocalVue();

describe('Notifikasi.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })

    test('render popup notification', () => {
        const VNotif = {
            props: ['notif'],
            template: '<div><slot :notif="notif" /></div>'
        }
        const wrapper = mount(Notifikasi, {
            localVue,
            vuetify,
            stubs: {
                VNotif
            }
        });
        expect(wrapper).toMatchSnapshot();
    })
    test('triggers options onAction click', async () => {
        const wrapper = mount(Notifikasi)
        await wrapper.trigger('click', { onAction: 0 })
    })
})
