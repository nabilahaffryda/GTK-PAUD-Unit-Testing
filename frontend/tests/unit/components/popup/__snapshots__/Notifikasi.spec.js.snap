// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifikasi.vue render popup notification 1`] = `
<div role="dialog" class="v-dialog__container">
  <!---->
</div>
`;
